lib/email_parser/parser.rb

```diff
@@ -375,5 +393,14 @@ def self.strip_subject(subject)
       stripped_subject
     end
     private_class_method(:strip_subject)
+
+    def self.remove_element_onwards(element)
+      return if element.nil?
+      while element.next_sibling
+        element.next_sibling.remove
```

Comment:
Is the `next_sibling` recomputed after removal? If not we would need to remove from the back.

Reply:
It was in my tests, but I'll verify

Reply:
confirmed:
```
2.6.1 :010 > doc = Nokogiri::HTML.parse('<div id="1"/><div id="2"/><div id="3"/><div id="4"/>')
 => #<Nokogiri::HTML::Document:0x3ff300c2b130 name="document" children=[#<Nokogiri::XML::DTD:0x3ff300c2af50 name="html">, #<Nokogiri::XML::Element:0x3ff300c2acf8 name="html" children=[#<Nokogiri::XML::Element:0x3ff300c2ab40 name="body" children=[#<Nokogiri::XML::Element:0x3ff300c2a988 name="div" attributes=[#<Nokogiri::XML::Attr:0x3ff300c2a924 name="id" value="1">]>, #<Nokogiri::XML::Element:0x3ff300c2a4ec name="div" attributes=[#<Nokogiri::XML::Attr:0x3ff300c2a488 name="id" value="2">]>, #<Nokogiri::XML::Element:0x3ff300c2a064 name="div" attributes=[#<Nokogiri::XML::Attr:0x3ff300859fd4 name="id" value="3">]>, #<Nokogiri::XML::Element:0x3ff300859bb0 name="div" attributes=[#<Nokogiri::XML::Attr:0x3ff300859b4c name="id" value="4">]>]>]>]> 
2.6.1 :011 > div = doc.xpath('//div').first
 => #<Nokogiri::XML::Element:0x3ff300c2a988 name="div" attributes=[#<Nokogiri::XML::Attr:0x3ff300c2a924 name="id" value="1">]> 
2.6.1 :012 > div.next_sibling
 => #<Nokogiri::XML::Element:0x3ff300c2a4ec name="div" attributes=[#<Nokogiri::XML::Attr:0x3ff300c2a488 name="id" value="2">]> 
2.6.1 :013 > div.next_sibling.remove
 => #<Nokogiri::XML::Element:0x3ff300c2a4ec name="div" attributes=[#<Nokogiri::XML::Attr:0x3ff300c2a488 name="id" value="2">]> 
2.6.1 :014 > div.next_sibling
 => #<Nokogiri::XML::Element:0x3ff300c2a064 name="div" attributes=[#<Nokogiri::XML::Attr:0x3ff300859fd4 name="id" value="3">]> 
```